src/consumer.rs

```diff
@@ -184,9 +202,57 @@ pub async fn from_substrait_rel(ctx: &mut SessionContext, rel: &Rel) -> Result<A
                 ))
             }
         }
+        Some(RelType::Aggregate(agg)) => {
+            if let Some(input) = agg.input.as_ref() {
+                let input = from_substrait_rel(ctx, input, extensions).await?;
+                let mut group_expr = vec![];
+                let mut aggr_expr = vec![];
+
+                let groupings = match agg.groupings.len() {
+                    1 => { Ok(&agg.groupings[0]) },
+                    _ => {
+                        Err(DataFusionError::NotImplemented(
```

Comment:
You could probably just `return Err` here and then no need to return an `Ok` for the single grouping?

Reply:
Actually, maybe this is fine. The `unwrap` after this can be replaced with `?`.